src/sagemaker/model.py

```diff
@@ -1375,13 +1375,17 @@ def deploy(
             sagemaker_session=self.sagemaker_session,
         )
 
-        tags = add_jumpstart_uri_tags(
-            tags=tags,
-            inference_model_uri=self.model_data
-            if isinstance(self.model_data, (str, dict))
-            else None,
-            inference_script_uri=self.source_dir,
-        )
+        if (
+            getattr(self.sagemaker_session, "settings", None) is not None
```

Comment:
same comment in `Estimator` class please

Reply:
This can only happen if someone creates a custom `Session`, and they explicitly put `None` for the `settings` field, rather than the default value. Since this would never happen in any normal case, I think it's okay to leave this as is. I actually had to do the `getattr` logic cause some of the unit tests mock the `Session` object so that the settings field is None, but this never happens in practice.